aircraft/wing/structure.py

```diff
@@ -0,0 +1,84 @@
+        tweb    = Variable('t_{web}', '-', 'Non-dim. shear web thickness')
+        Wcap    = Variable('W_{cap}', 'N', 'Weight of spar caps')
+        Wweb    = Variable('W_{web}', 'N', 'Weight of shear web')
+        Wstruct = Variable('W_{struct}', 'N', 'Structural weight')
+
+        # Constants
+        fwadd  = Variable('f_{w,add}', fwadd, '-', 'Wing added weight fraction') # TASOPT code (737.tas)
+        g      = Variable('g', 9.81, 'm/s^2', 'Gravitational acceleration')
+        Nlift  = Variable('N_{lift}', Nlift, '-', 'Wing loading multiplier')
+        rh     = Variable('r_h', rh, '-', 'Fractional wing thickness at spar web')
+        rhocap = Variable('\\rho_{cap}', rhocap, 'kg/m^3', 'Density of spar cap material' )
+        rhoweb = Variable('\\rho_{web}', rhoweb, 'kg/m^3', 'Density of shear web material')
+        sigmax = Variable('\\sigma_{max}', sigmax, 'Pa', 'Allowable tensile stress')
+        sigmaxshear = Variable('\\sigma_{max,shear}', sigmaxshear, 'Pa', 'Allowable shear stress')
+        w      = Variable('w', w, '-', 'Wingbox-width-to-chord ratio')
+
+        objective = Wstruct
+
+        constraints = [
+                       # Aspect ratio definition
+                       A == b**2/S,
+
+                       # Defining taper dummy variables
+                       p >= 1 + 2*taper,
+                       2*q >= 1 + p,
+
+                       # Upper bound on maximum thickness
+                       tau <= 0.15,
+
+                       # Root moment calculation (see Hoburg 2014)
+                       # Depends on a given load the wing must support, Lmax
+                       # Assumes net lift per unit span proportional to the local chord
+                       Mr >= Lmax*A*p/24,
+
+                       # Root stiffness (see Hoburg 2014)
+                       # Assumes rh = 0.75, so that rms box height = ~0.92*max thickness
+                       0.92*w*tau*tcap**2 + Icap <= 0.92**2/2*w*tau**2*tcap,
+
+                       # Stress limit
+                       # Assumes bending stress carried by caps only (Icap >> Iweb)
+                       8 >= Nlift*Mr*A*q**2*tau/(S*Icap*sigmax),
+
+                       # Shear web sizing
+                       # Assumes all shear loads are carried by web and rh = 0.75
+                       12 >= A*Lmax*Nlift*q**2/(tau*S*tweb*sigmaxshear),
+
+                       # Posynomial approximation of nu = (1+lam+lam**2)/(1+lam**2)
+                       nu**3.94 >= 0.86*p**(-2.38)+ 0.14*p**0.56,
+
+                       # Weight of spar caps and shear webs
+                       Wcap >= 8*rhocap*g*w*tcap*S**1.5*nu/(3*A**0.5),
+                       Wweb >= 8*rhoweb*g*rh*tau*tweb*S**1.5*nu/(3*A**0.5),
+
+                       # Total wing weight using an additional weight fraction
+                       Wstruct >= fwadd*(Wweb + Wcap),
+                       ]
+        return objective, constraints
```

Comment:
The model is not really standalone because either Aspect Ratio or Span needs to be lower bounded/have upward pressure, and that comes from an aerodynamics model, not a structural model. This is pretty relevant to the conversation we just had.